tests/test_routes.py

```diff
@@ -43,3 +47,107 @@ def test_index(self):
         """ It should call the home page """
         resp = self.client.get("/")
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
+
+    def test_create_customer(self):
+        """ Create a new Customer """
+        # Arrange
+
+
+        # customer = CustomerFactory()
+        # logging.debug(customer)
+        # customer.id = None
+        # customer.create()
+        # self.assertIsNotNone(customer.id)
+        # # Fetch it back
+        # found_customer = Customer.find(customer.id)
+        # self.assertEqual(found_customer.id, customer.id)
+        # self.assertEqual(found_customer.first_name, customer.first_name)
+        # self.assertEqual(found_customer.last_name, customer.last_name)
+        # self.assertEqual(found_customer.email, customer.email)
+        # self.assertEqual(found_customer.address, customer.address)
+
+        fake_customer = CustomerFactory()
+        data = json.dumps({
+            "id": fake_customer.id,
+            "first_name": fake_customer.first_name,
+            "last_name": fake_customer.last_name,
+            "email": fake_customer.email,
+            "address": fake_customer.address
+        })
+
+        # Action
+        response = self.client.post(
+            "/customers", 
+            data=data,
+            content_type="application/json"
+        )
+
+        # Assert
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        
+        # Check the data is correct
+        new_json = json.loads(response.data)
+        self.assertEqual(new_json["first_name"], fake_customer.first_name)
+        self.assertEqual(new_json["last_name"], fake_customer.last_name)
+        self.assertEqual(new_json["email"], fake_customer.email)
+        self.assertEqual(new_json["address"], fake_customer.address)
+
+    def test_create_customer_not_json(self):
+        """ Create a new Customer with non-JSON data """
+        # Arrange
+        fake_customer = CustomerFactory()
+
+        # Convert the data to a string instead of JSON
+        data = str({
```

Comment:
please call Customer.serialize() instead of str(), i.e. data = fake_customer.serialize()